File: CS/03_Slides/GetSlideByIndexOrID.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;

namespace GetSlideByIndexOrID
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
             //Create a PPT document
            Presentation presentation = new Presentation();

            //Load document from disk
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\BlankSample_N.pptx");

            //Get slide by index 0
            ISlide slide1 = presentation.Slides[0];
            //Append a shape in the slide
            IAutoShape shape1=slide1.Shapes.AppendShape(ShapeType.Rectangle, new RectangleF(100, 100, 200, 100));
            //Add text in the shape
            shape1.TextFrame.Text = "Get slide by index";

            //Get slide by slide ID
            ISlide slide2 = presentation.FindSlide((int)presentation.Slides[1].SlideID);
            //Append a shape in the slide
            IAutoShape shape2 = slide2.Shapes.AppendShape(ShapeType.Rectangle, new RectangleF(100, 100, 200, 100));
            //Add text in the shape
            shape2.TextFrame.Text = "Get slide by slide id";

            //Save the document
            string result = "GetSlideByIndexOrID_result.pptx";
            presentation.SaveToFile(result, FileFormat.Pptx2013);

            //Launch the file
            OutputViewer(result);
        }
        private void OutputViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/03_Slides/GetSlideByIndexOrID.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation

Namespace GetSlideByIndexOrID
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			 'Create a PPT document
			Dim presentation As New Presentation()

			'Load document from disk
			presentation.LoadFromFile("..\..\..\..\..\..\Data\BlankSample_N.pptx")

			'Get slide by index 0
			Dim slide1 As ISlide = presentation.Slides(0)
			'Append a shape in the slide
			Dim shape1 As IAutoShape=slide1.Shapes.AppendShape(ShapeType.Rectangle, New RectangleF(100, 100, 200, 100))
			'Add text in the shape
			shape1.TextFrame.Text = "Get slide by index"

			'Get slide by slide ID
			Dim slide2 As ISlide = presentation.FindSlide(CInt(Fix(presentation.Slides(1).SlideID)))
			'Append a shape in the slide
			Dim shape2 As IAutoShape = slide2.Shapes.AppendShape(ShapeType.Rectangle, New RectangleF(100, 100, 200, 100))
			'Add text in the shape
			shape2.TextFrame.Text = "Get slide by slide id"

			'Save the document
			Dim result As String = "GetSlideByIndexOrID_result.pptx"
			presentation.SaveToFile(result, FileFormat.Pptx2013)

			'Launch the file
			OutputViewer(result)
		End Sub
		Private Sub OutputViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace